pkg/store/bucket.go

```diff
@@ -885,7 +886,7 @@ func populateChunk(out *storepb.AggrChunk, in chunkenc.Chunk, aggrs []storepb.Ag
 		if err != nil {
 			return err
 		}
-		out.Raw = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b}
+		out.Raw = &storepb.Chunk{Type: storepb.Chunk_XOR, Data: b, Hash: xxhash.Sum64(b)}
```

Comment:
Let's add a flag to this https://github.com/thanos-io/thanos/blob/main/pkg/store/storepb/rpc.proto#L88?
Since store gateway is also used by other projects like Cortex, it is better to control this behavior instead of enabling it by default.